writer.go

```diff
@@ -443,7 +443,7 @@ func (p *MediaPlaylist) Encode() *bytes.Buffer {
 
 	head := p.head
 	count := p.count
-	for i := uint(0); i < p.winsize && count > 0; count-- {
+	for i := uint(0); (i < p.winsize || p.winsize == 0) && count > 0; count-- {
```

Comment:
Actually, regardless of where the underlying issue it, it appears the docs require `winsize` to be 0 for VoD: https://github.com/grafov/m3u8/blob/11af315b1d436b4b501d34b0c7b09323130a089f/structure.go#L105

So I think your original proposal is likely the best option for backwards compatibility.